### PR TITLE
Adds small chance to spawn intact helicopter at military helipad

### PR DESCRIPTION
**Pull request by CountAlex**
#### Summary
SUMMARY: Balance "Adds small chance to spawn intact helicopter at military helipad"

#### Purpose of change
A chance to utilize helicopter piloting skills with a really small chance to find working helicopter at military helipad.

#### Describe the solution
Simply adding it to mapgen

#### Testing
Spawning military helipad with 100% chance of seeing Apache there, checking if it spawned and placed properly.

#### Additional context
Devs might want to see another spawn chance than 1% which will be adjusted accordingly then.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
1% chance is nothing. There ain't so many helicopter pilots out there, so 10% or even 20% should be good and balanced.

**Comment by CountAlex** [comment]
> 
> 
> 1% chance is nothing. There ain't so many helicopter pilots out there, so 10% or even 20% should be good and balanced.

I have similar view but Kevin in Discord conversation pointed that it should be pointed towards realistic distribution in which case probability to find one is <1%. If you think it would be approved with higher chance, I will adjust spawn chance to 10-20%.

**Comment by Night-Pryanik** [comment]
I guess that chance to find intact helicopter _somewhere_ could be less than one percent, that's ok. But we're talking about _military helipad_, so ten percent chance feels like ok here.
But anyway, let's hear Kevin.

**Comment by CountAlex** [comment]
@kevingranade Please weigh in.

**Comment by kevingranade** [comment]
As you said, I already answered the question, why are you asking again?

**Comment by CountAlex** [comment]
> 
> 
> As you said, I already answered the question, why are you asking again?

Didn't mean to bother you, but you can see above a different opinion from a Dev member while I am not. Thanks for answering.

**Comment by kevingranade** [comment]
Oh it's specifically an Apache? nope, nope, nope, so much nope.
One, I'm super sceptical that even an ordinary helicopter pilot is going to be able to handle an apache, it's a whole different beast.
Two, I'm sceptical that an apache sitting on the tarmac is going to be either flyable or repairable to fly given survivor resources.
Three, the numbers are even more upside down than for ordinary military helicopters, there have been 2,400 Apache helicopters made *ever*, finding one at all, much less in working order is beyond any reasonable expectation.

**Comment by CountAlex** [comment]
Replaced helicopter to small 2-seater, adjusted spawning chance.

**Comment by kevingranade** [comment]
I actually did find a better source for current military aircraft
https://en.wikipedia.org/wiki/List_of_active_United_States_military_aircraft
and it's not *quite* what I expected.
Here's the tl;dr for Army numbers:

Name | Manufacturer | Origin | Class | Role | Introduced | In Service
-- | -- | -- | -- | -- | -- | --
MH/AH-6M Little Bird | MD Helicopter | USA | Helicopter | Attack | 1980 | 47
AH-64 Apache Longbow, Guardian | Boeing | USA | Helicopter | Attack | 1986 | 792
CH-47 Chinook | Boeing | USA | Helicopter | Transport | 1962 | 442
EH-60A Black Hawk | Sikorsky | USA | Helicopter | Electronic-warfare | 1979 | 64
MH-47 Chinook | Boeing | USA | Helicopter | Multi-mission | 1962 | 61
MH-60 Black Hawk | Sikorsky | USA | Helicopter | Multi-mission |   | 58
TH-67 Creek | Bell | USACanada | Helicopter | Trainer |   | 88
UH-60 Black Hawk | Sikorsky | USA | Helicopter | Utility |   | 1,443
UH-72A Lakota | Eurocopter | USAGermany | Helicopter | Utility |   | 341

So the Blackhawk dominates, but the Apache is a looot closer than I thought it would be, and the Chinook and Lakota bring up the rear, everything else is too rare to really bother with.
I guess the "number produced" skews hard toward the helicopters produced for (and were destroyed in) the Vietnam and Korean wars.

So what I'm thinking is, have those 4 spawn in those ratios, and each of them gets a chunk of that 1%, so roughly 0.5% Blackhawks, 0.25% Apaches, 0.15% Chinook and 0.1% Lakota.

If we don't have representative vehicles for them, I guess pull those out and redistribute? Really what we want is a sizeable transport of some kind taking up most of the slots, and the Apache taking that 0.25%.

**Comment by CountAlex** [comment]
Adjusting to that.

File: data/json/mapgen/helipad.json [hunk 1]
```diff
@@ -122,8 +122,8 @@
         { "vehicle": "forklift", "x": 29, "y": 6, "chance": 95, "rotation": 180, "status": 1 },
         { "vehicle": "parkinglotbasic", "x": 4, "y": 9, "chance": 75, "rotation": 270, "status": -1 },
         { "vehicle": "flatbed_truck", "x": 11, "y": 4, "chance": 25, "rotation": 270, "status": -1 },
-        { "vehicle": "SmallFlier3_Wreck", "x": 9, "y": 38, "chance": 75, "rotation": 90, "status": -1 },
-        { "vehicle": "2seater2", "x": 24, "y": 33, "chance": 25, "rotation": 270, "status": 1 },
+        { "vehicle": "Medevac1", "x": 9, "y": 38, "chance": 1, "rotation": 90, "status": -1 },
+        { "vehicle": "helicopter_apache_1a", "x": 24, "y": 1, "chance": 25, "rotation": 270, "status": 1 },
```
Comment (Wishbringer) [inline data/json/mapgen/helipad.json:126]:
```suggestion
        { "vehicle": "helicopter_apache_1a", "x": 24, "y": 33, "chance": 1, "rotation": 270, "status": 1 },
```
Did you mean to do this?

Comment (CountAlex) [reply data/json/mapgen/helipad.json]: Yes indeed!

**Pull request merged**